WordPress/src/main/java/org/wordpress/android/util/image/ImageManager.kt

```diff
@@ -142,6 +200,22 @@ class ImageManager @Inject constructor(val placeholderManager: ImagePlaceholderM
         return if (errorImageRes == null) this else this.error(errorImageRes)
     }
 
+    private fun GlideRequest<Drawable>.addThumbnail(
+        context: Context,
+        thumbnailUrl: String?,
+        listener: RequestListener<Drawable>
+    ): GlideRequest<Drawable> {
+        return if (TextUtils.isEmpty(thumbnailUrl)) {
+            this
```

Comment:
Not stupid at all... Kotlin has many secrets ;). 

Glide uses Builder pattern -> so each method returns GlideRequest and you can call another method on the result -> therefore you can easily chain "operators" (eg. `GlideApp.load(..).centerCrop().resize(..).thumbnail(..)....` ). I wanted to create another function which would be chainable, so I created a private <strong>extension</strong> function for `GlideRequest`. Since it's an extension function, `this` refers to an instance of `GlideRequest`. Hope that helps;).

Reply:
Thanks for the explanation! Makes sense now :)